docs/design/architectural-overview.md

```diff
@@ -0,0 +1,168 @@
+Kuadrant defines a [gateway class](https://gateway-api.sigs.k8s.io/api-types/gatewayclass/) that the multi-cluster gateway controller is the specified controller for. An OCM user can then (with the correct RBAC in place) define a Gateway just as they would in a single cluster setting and this will be reconciled by the multi-cluster gateway controller. We use this gateway definition to represent a "multi-cluster" enabled gateway and use the aggregated status from each instance of this gateway to store information such as IPs and Hostnames of Load Balancers and how many routes are attached to a given listner.
+
+![](./images/ocm-integration.png)
+
+
+### Gateway Deployment and Distribution
+
+In order for a multi-cluster gateway to be useful, it needs to be distributed or "placed" on a specific set of hub managed spoke clusters.  OCM is responsible for a set of placement and replication APIs. Kuadrant is aware of these APIs, and so when a given gateway is chosen to be placed on a set of managed clusters, Kuadrant multi-cluster gateway controller will ensure the right resources ([ManifestWork](https://open-cluster-management.io/concepts/manifestwork/)) are created in the correct namespaces in the hub. OCM then is responsible for syncing these to the actual spoke cluster and reporting back the status of these resources to the Hub. 
+
+In order for the Gateway to be instantiated, we need to know what gateway provider is being used on the spoke clusters. Admins can then set this provider in the hub via the Gateway class params. Kuadrant will then apply a transformation to ensure the gateway being synced references this spoke gateway provider (istio for example). 
+
+It is the OCM workagent that is responsible for syncing down and applying the resources into the managed spoke cluster. It is also responsible for syncing status information back to the hub. 
+
+The status information reported back to the Hub is used by the multi-cluster gateway controller to know what LB hosts / IPAddresses to use for DNSRecords that it creates and manages.
+
+More info can be found here https://open-cluster-management.io/concepts/architecture/ 
+
+## How does Kuadrant integrate with Gateway Providers
+
+The Kuadrant data plane, integrates with the Istio gateway provider only currently: 
+
+- It registers Authorino with the `IstioOperator` as an auth provider.
+
+- It leverages an `EnvoyFilter`  to register the rate limiting service as a upstream service.
+
+- It leverages istio `AuthorizationPolicy` to configure when Authorino should be called for a given host and path etc. 
+
+- It provides a WASM Plugin that conforms to the Proxy WASM ABI. This WASM Plugin is configured via the Kuadrant Operator based on valid rate limit policies and is then executed by the underlying Envoy instance that provides the proxy layer for the Gateway (IE Envoy). This plugin currently manages when RateLimiting and knows the location of the Limitador service.
```

Comment:
"Manages when RateLimiting" and knows the location..."

I can't parse that sentence.